activerecord/lib/active_record/reflection.rb

```diff
@@ -378,6 +378,15 @@ def autosave=(autosave)
       # instead. This allows plugins to hook into association object creation.
       def klass
         @klass ||= compute_class(class_name)
+
+        unless @klass.respond_to?(:table_name)
```

Comment:
Not sure if we should use `@klass.is_a?(Class)` instead since when we get a namespace candidate we will pick an object that is not a `Class` but a `Module`